src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -205,10 +235,75 @@ export class ExamExerciseImportComponent implements OnInit {
             // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
             exercise.shortName?.length! > 2 &&
             this.shortNamePattern.test(exercise.shortName!) &&
+            !this.exercisesWithDuplicatedShortNames.has(exercise.id!) &&
             exercise.shortName !== this.getBlocklistShortNameOfProgrammingExercise(exercise.id!)
         );
     }
 
+    /**
+     * checks if there are any selected exercises with the same title or short name as the passed exercise
+     * @param exercise     exercise we want to use to check for duplications
+     * @param exerciseGroup exercise group of the exercise
+     * @param checkForTitle true if the title should be checked, otherwise the short name is checked
+     */
+    checkForDuplicatedTitlesOrShortNames(exercise: Exercise, exerciseGroup: ExerciseGroup, checkForTitle: boolean) {
+        if (!this.exerciseIsSelected(exercise, exerciseGroup)) {
+            return;
+        }
+
+        const duplicatesToCheck = checkForTitle ? this.exercisesWithDuplicatedTitles : this.exercisesWithDuplicatedShortNames;
+        let hasDuplicate = false;
+        this.selectedExercises.forEach((exerciseGroup) => {
+            exerciseGroup.forEach((ex) => {
+                if (ex.type === ExerciseType.PROGRAMMING && ex !== exercise) {
+                    if ((checkForTitle && ex.title === exercise.title) || (!checkForTitle && ex.shortName === exercise.shortName)) {
+                        hasDuplicate = true;
+                        duplicatesToCheck.set(ex.id!, checkForTitle ? exercise.title! : exercise.shortName!);
+                    }
+                }
+            });
+        });
+
+        //check if the exercise was a duplicate before
+        const titleOrShortName = duplicatesToCheck.get(exercise.id!);
+        if (duplicatesToCheck.delete(exercise.id!)) {
+            this.removeExerciseFromDuplicates(titleOrShortName!, checkForTitle);
+        }
+
+        if (hasDuplicate) {
+            duplicatesToCheck.set(exercise.id!, checkForTitle ? exercise.title! : exercise.shortName!);
+        }
+    }
+
+    /**
+     * after a duplicate programming exercise's title / short name has been edited or a programming exercise has been unselected,
+     * other duplicates need to be checked if they can be removed from the map. I case there is only one exercise with the same title / short name
```

Comment:
```suggestion
     * other duplicates need to be checked if they can be removed from the map. In case there is only one exercise with the same title / short name,
```